### DISCUSSION TITLE
Add  "SEPLOS" 200A BMS

### DISCUSSION THREAD

**Discussion by ornolio** [fl4p/batmon-ha#136] [Ideas]
What about add support for Lifepo4 "SEPLOS" 200A BMS Bluetooth interface on BATMON?

[SEPLOS 16S(LFP) 200A BMS SPECIFICATION V16.pdf](https://www.seplos.com/download.html?page=2&limit=10)

**Comment by fl4p** [comment]
No progress on this one yet.

issue https://github.com/fl4p/batmon-ha/issues/52

**Reply by rira2005** [reply]
Yes pls this one is a upcomming BMS! i would pay a fee to add it!
TIA

**Reply by linuxdep** [reply]
Yes, that would solve the problem if you use several Seploas BMS in series, then the master is invisible. 
 At the moment there is probably no solution via BT.

**Reply by saviothecnic** [reply]
It would make me happy too if the Seplos BMS was supported
Seplos was the first BMS to come out to manage the parallel
And so there are a lot of them around and installed

**Reply by ornolio** [reply]
check syssi/esphome-seplos-bms on github for the esp32 ble version.
it works like a charm: this is for a single bms, but I easily  extended it  to 4 bms seplos v2 in a stack.

https://github.com/syssi/esphome-seplos-bms/blob/main/esp32-ble-example.yaml

**Reply by fl4p** [reply]
batmon now support seplos BMS

**Reply by saviothecnic** [reply]
> batmon now support seplos BMS

From which version does it support it?
I tried this configuration yesterday,
but it still only showed DalyBMS.

- address: 02:11:23:xx:xx:xx
  type: daly
  alias: DalyBms
- address: 60:6E:41:xx:xx:xx
  type: seplosv2
  alias: Seplos1
- address: 60:6E:41:xx:xx:xx
  type: seplosv2
  alias: Seplos2
- address: 60:6E:41:xx:xx:xx
  type: seplosv2
  alias: Seplos3

**Comment by ornolio** [comment]
I agree.

**Comment by itiotdk** [comment]
i have the bluetooth protocol from seplos if that helps.  got it from seplos after asking some times... where can i send it?  - Andy

**Reply by fl4p** [reply]
Hi Andy, yes it helps. you can attach it here

**Comment by itiotdk** [comment]
[Seplos bluetooth protocol 2.0.docx](https://github.com/fl4p/batmon-ha/files/14386074/Seplos.bluetooth.protocol.2.0.docx)
here you are.

**Reply by fl4p** [reply]
thanks! i'll have a look

**Reply by fl4p** [reply]
[Seplos.bluetooth.protocol.2.pdf](https://github.com/fl4p/batmon-ha/files/14391802/Seplos.bluetooth.protocol.2.pdf)
same document as pdf

**Comment by fl4p** [comment]
ble code
https://github.com/syssi/esphome-seplos-bms/pull/52/commits/e65ef7a2334b4a910790aee9807177ba8c048a75#diff-1b095da13c3c6231c14dd2f234bc78eaa159b2f8e4402b384286bdb6d55cf3ab

**Reply by itiotdk** [reply]
I gave the doc to Sussi as well. But I did not get any feedback. Did he use it?  

I would love to somehow read data out from Bluetooth on the seplos as the BMS can’t be monitored by cable when in can bus mode and connected to my deye inverter.

**Comment by ornolio** [comment]
Attached is a file of read data out from Seplos v.2 RS485 port (not CAN) through RS485-USB converter.
data have been extracted using Seplos APP, but they are not raw data, instead are formatted in a log by the app.
Is that what are you looking for? Can it help you?

Regards

**Comment by ornolio** [comment]
file mentioned
[DataRecord_20240201_034205.xls](https://github.com/fl4p/batmon-ha/files/14401675/DataRecord_20240201_034205.xls)

**Comment by andreas-bulling** [comment]
Any update on this? It would be great to have Seplos support

**Reply by saviothecnic** [reply]
Yes, I'm waiting too, it would be really fantastic
have Seplos BMS support in Bluetooth
together with the Daly they are the most used BMS

**Comment by bpc80** [comment]
Push. PLEASE Add Seplos BMS. This would make many people happy.

**Comment by Goaheadz** [comment]
up

**Comment by Goaheadz** [comment]
There is already one integration working with bluetooth and Seplos.

Check it.
https://github.com/patman15/BMS_BLE-HA

**Reply by saviothecnic** [reply]
> There is already one integration working with bluetooth and Seplos.
> Check it. https://github.com/patman15/BMS_BLE-HA
I'll check
but it seems to me that this doesn't send data in MQTT
like Batmon does, which I already use for the Daly BMS.
And it would be nice not to add any other integrations to
also see the Seplos BMS in BT.

**Comment by saviothecnic** [comment]
Has Seplos BMS V2 support via Bluetooth been added?

**Comment by andreas-bulling** [comment]
I don't think so. Just use the integration mentioned above. Works great!

**Reply by saviothecnic** [reply]
Yes, I'm trying it, but it doesn't send data in MQTT
and it has many fewer sensors than this one.

It doesn't have the sensors for the individual 16 cells and many others.
It works well with BT integration
which happens automatically
and it sees Seplos V2 BMS without problems.
But it has many limitations compared to BatMon,
which from what I see on DalyBms is much more complete.
I'm attaching what I see with the two integrations so you can understand that this one is indeed much more complete, in addition to having native MQTT data.

DalyBMS BatMon
<img width="414" height="1355" alt="image" src="https://github.com/user-attachments/assets/b14c9317-87cd-4b51-a962-1002bbe71b14" />
 
 Seplos V2 [BMS_BLE-HA
<img width="641" height="660" alt="image" src="https://github.com/user-attachments/assets/60300f40-b550-4b5a-89be-751d9e9aea1c" />

**Comment by ornolio** [comment]
substitutions:
  name: seplos-bms-ble
  device_description: "Monitor a Seplos Battery Management System via BLE"
  external_components_source: ***@***.***
  mac_address: 90:A6:BF:93:A0:69

esphome:
  name: ${name}
  comment: ${device_description}
  min_version: 2024.6.0
  project:
    name: "syssi.esphome-seplos-bms"
    version: 1.0.0

esp32:
  board: wemos_d1_mini32
  framework:
    type: esp-idf

external_components:
  - source: ${external_components_source}
    refresh: 0s

wifi:
  ssid: !secret wifi_ssid
  password: !secret wifi_password

ota:
  platform: esphome
  on_begin:
    then:
      - switch.turn_off: ble_client_switch0
      - logger.log: "BLE connection suspended for OTA update"

logger:
  level: DEBUG

# If you don't use Home Assistant please remove this `api` section and uncomment the `mqtt` component!
api:
  reboot_timeout: 0s

# mqtt:
#   broker: !secret mqtt_host
#   username: !secret mqtt_username
#   password: !secret mqtt_password
#   id: mqtt_client

esp32_ble_tracker:

ble_client:
  - mac_address: ${mac_address}
    id: client0

seplos_bms_ble:
  - ble_client_id: client0
    id: bms0
    update_interval: 10s

binary_sensor:
  - platform: seplos_bms_ble
    seplos_bms_ble_id: bms0
    charging:
      name: "${name} charging"
    discharging:
      name: "${name} discharging"
    limiting_current:
      name: "${name} limiting current"
    online_status:
      name: "${name} online status"

sensor:
  - platform: seplos_bms_ble
    seplos_bms_ble_id: bms0
    voltage_protection_bitmask:
      name: "${name} voltage protection bitmask"
    current_protection_bitmask:
      name: "${name} current protection bitmask"
    temperature_protection_bitmask:
      name: "${name} temperature protection bitmask"
    alarm_event1_bitmask:
      name: "${name} alarm event 1 bitmask"
    alarm_event2_bitmask:
      name: "${name} alarm event 2 bitmask"
    alarm_event3_bitmask:
      name: "${name} alarm event 3 bitmask"
    alarm_event4_bitmask:
      name: "${name} alarm event 4 bitmask"
    alarm_event5_bitmask:
      name: "${name} alarm event 5 bitmask"
    alarm_event6_bitmask:
      name: "${name} alarm event 6 bitmask"
    alarm_event7_bitmask:
      name: "${name} alarm event 7 bitmask"
    alarm_event8_bitmask:
      name: "${name} alarm event 8 bitmask"
    total_voltage:
      name: "${name} total voltage"
    current:
      name: "${name} current"
    power:
      name: "${name} power"
    charging_power:
      name: "${name} charging power"
    discharging_power:
      name: "${name} discharging power"
    capacity_remaining:
      name: "${name} capacity remaining"
    state_of_charge:
      name: "${name} state of charge"
    nominal_capacity:
      name: "${name} nominal capacity"
    charging_cycles:
      name: "${name} charging cycles"
    min_cell_voltage:
      name: "${name} min cell voltage"
    max_cell_voltage:
      name: "${name} max cell voltage"
    min_voltage_cell:
      name: "${name} min voltage cell"
    max_voltage_cell:
      name: "${name} max voltage cell"
    delta_cell_voltage:
      name: "${name} delta cell voltage"
    average_cell_voltage:
      name: "${name} average cell voltage"
    average_cell_temperature:
      name: "${name} average cell temperature"
    ambient_temperature:
      name: "${name} ambient temperature"
    mosfet_temperature:
      name: "${name} mosfet temperature"
    state_of_health:
      name: "${name} state of health"
    port_voltage:
      name: "${name} port voltage"
    battery_capacity:
      name: "${name} battery capacity"
    temperature_1:
      name: "${name} temperature 1"
    temperature_2:
      name: "${name} temperature 2"
    temperature_3:
      name: "${name} temperature 3"
    temperature_4:
      name: "${name} temperature 4"
    cell_voltage_1:
      name: "${name} cell voltage 1"
    cell_voltage_2:
      name: "${name} cell voltage 2"
    cell_voltage_3:
      name: "${name} cell voltage 3"
    cell_voltage_4:
      name: "${name} cell voltage 4"
    cell_voltage_5:
      name: "${name} cell voltage 5"
    cell_voltage_6:
      name: "${name} cell voltage 6"
    cell_voltage_7:
      name: "${name} cell voltage 7"
    cell_voltage_8:
      name: "${name} cell voltage 8"
    cell_voltage_9:
      name: "${name} cell voltage 9"
    cell_voltage_10:
      name: "${name} cell voltage 10"
    cell_voltage_11:
      name: "${name} cell voltage 11"
    cell_voltage_12:
      name: "${name} cell voltage 12"
    cell_voltage_13:
      name: "${name} cell voltage 13"
    cell_voltage_14:
      name: "${name} cell voltage 14"
    cell_voltage_15:
      name: "${name} cell voltage 15"
    cell_voltage_16:
      name: "${name} cell voltage 16"

switch:
  - platform: seplos_bms_ble
    seplos_bms_ble_id: bms0
    discharging:
      name: "${name} discharging"
    charging:
      name: "${name} charging"
    current_limit:
      name: "${name} current limit"
    heating:
      name: "${name} heating"

  - platform: ble_client
    ble_client_id: client0
    name: "${name} enable bluetooth connection"
    id: ble_client_switch0

text_sensor:
  - platform: seplos_bms_ble
    seplos_bms_ble_id: bms0
    software_version:
      name: "${name} software version"
    device_model:
      name: "${name} device model"
    hardware_version:
      name: "${name} hardware version"
    voltage_protection:
      name: "${name} voltage protection"
    current_protection:
      name: "${name} current protection"
    temperature_protection:
      name: "${name} temperature protection"
    alarms:
      name: "${name} alarms"


from syssi/esphome-seplos-bms on github. check for the esp32 ble version.
it works like a charm: this is for a si gle bms, but I extended it it to 4 bms seplos v2 in a stack.

ciao.
--
Inviato da Libero Mail

**Reply by saviothecnic** [reply]
This is great news.
So I can use just one ESP32
to integrate up to 4 Seplos V2 BMSs in BT?

Do you have the code for 4 BMSs so I can try it? I should replace
only the BT MAC addresses of my BMSs in your code. Is that correct?

**Comment by ornolio** [comment]
Thank you Fabian. I didn't know, is it a recent upgrade?
I asked for seplos support since 2024. Well I'll try it asap.
Does it support single bms or multiple bms in a daisy chain?

Best Regards.
--
Inviato da Libero Mail

**Reply by fl4p** [reply]
check https://github.com/fl4p/batmon-ha?tab=readme-ov-file#supported-devices-bluetooth-low-energy

Not sure, I don't own a Seplos, you can just try and let me know

**Comment by ornolio** [comment]
Yes you have only to replace the ble address on top of the yaml and install esphome on ha, if you do not have yet.

Esp32 diesn't have any pin connected, just the power.

Please consider that Fabian wrote me today that now batmon support seplos version 2 bms. Please check the Batmon setup.

Next week I'll add my yaml version supporting multiple seplos bms on the github within a discussion of the syssi/esphome-seplos-bms.

Please check there. 
--
Inviato da Libero Mail

**Comment by ornolio** [comment]
As promised, pleasee check this discusion on github: https://github.com/syssi/esphome-seplos-bms/discussions/163#top 

Title: "extend https://github.com/syssi/esphome-seplos-bms/blob/main/esp32-ble-example.yaml to 4 vers. 2 seplos bms #163"

to have the 4 bms ble version.
